core/trino-main/src/main/java/io/trino/sql/planner/optimizations/BeginTableWrite.java

```diff
@@ -267,9 +268,13 @@ private TableHandle findTableScanHandle(PlanNode node)
             }
             if (node instanceof JoinNode) {
                 JoinNode joinNode = (JoinNode) node;
-                if (joinNode.getType() == JoinNode.Type.INNER && isAtMostScalar(joinNode.getRight())) {
```

Comment:
nit: I would love to see in the commit description why this check is no longer needed :)

Reply:
👍 Why this check is removed?

Reply:
Now with support for correlated queries for assignment...we would translate them to a `LEFT JOIN` with a non SCALAR build source.

Reply:
moved to top level discussion @ https://github.com/trinodb/trino/pull/8286#issuecomment-865113772